apps/coscrad-frontend/src/components/higher-order-components/categorizable-page-layout.tsx

```diff
@@ -0,0 +1,174 @@
+                        <HubIcon />
+                    </IconButton>
+                </Tooltip>
+            </Box>
+            <Drawer
+                anchor="right"
+                PaperProps={{
+                    sx: { width: '35vw' },
+                }}
+                variant="temporary"
+                data-testid="connected-resources-panel"
+                open={drawerState['right']}
+            >
+                <DrawerHeader>
+                    <Box sx={{ mb: 2 }}>
+                        <Typography variant="h3" sx={{ mb: 0 }}>
+                            Connected Resources
+                        </Typography>
+                        <Typography variant="subtitle2">
+                            {/* TODO: Remove this in future, it is for troubleshooting purposes only */}
+                            for {resourceType}/{id}
+                        </Typography>
+                    </Box>
+                    <Tooltip title="Close Panel">
+                        <IconButton
+                            data-testid="close-connected-resources-panel-button"
+                            onClick={() => {
+                                toggleDrawer('right', false);
+                            }}
+                        >
+                            <CloseIcon />
+                        </IconButton>
+                    </Tooltip>
+                </DrawerHeader>
+                <Divider variant="fullWidth" sx={{ mb: 3 }} />
+                <DrawerContentStack>{connectedResourcesList}</DrawerContentStack>
+            </Drawer>
+            <Drawer
+                anchor="bottom"
+                PaperProps={{
+                    sx: {
+                        height: isBottomDrawerExpanded ? '80vh' : '40vh',
+                        width: '90vw',
+                        margin: 'auto',
+                        borderLeft: '1px solid rgba(0, 0, 0, 0.12)',
+                        borderRight: '1px solid rgba(0, 0, 0, 0.12)',
+                        borderRadius: '20px 20px 0px 0px',
+                    },
+                }}
+                variant="temporary"
+                data-testid="notes-panel"
+                open={drawerState['bottom']}
+            >
+                <DrawerHeader>
+                    <Box sx={{ mb: 2 }}>
+                        <Typography variant="h3" sx={{ mb: 0 }}>
+                            Notes
+                        </Typography>
+                        <Typography variant="subtitle2">
+                            for {resourceType}/{id}
+                        </Typography>
+                    </Box>
+                    <Box>
+                        <Tooltip title="Expand/Contract Notes Panel">
+                            <IconButton
+                                data-testid="expand-notes-panel-button"
+                                onClick={() => {
+                                    setIsBottomDrawerExpanded(!isBottomDrawerExpanded);
```

Comment:
This component has become very complex. If you are going to have an expand \ contract behaviour (I would not have added this here at first pass if I'm trying to do things iteratively), you should break each panel \ drawer into it's own component and encapsulate the expanded \ contracted state there.